### ISSUE TITLE
[BUG] Can't load a dump file using WPDb::importSqlDumpFile, driver is not loaded

### ISSUE THREAD

**Issue by andergmartins** [lucatume/wp-browser#557]
**Environment**
OS: Linux
PHP version: 7.4
Installed Codeception version: 4.1.27
Installed wp-browser version: 3.0.18
WordPress version: 5.8.3
Local development environment: Docker/DevKinsta
WordPress structure and management: default

**Codeception configuration file**  

```yaml
paths:
  tests: tests/codeception
  output: tests/codeception/_output
  data: tests/codeception/_data
  support: tests/codeception/_support
  envs: tests/codeception/_envs
actor_suffix: Tester
extensions:
  enabled:
    - Codeception\Extension\RunFailed
    - tad\WPBrowser\Extension\Copier
  commands:
    - Codeception\Command\GenerateWPUnit
    - Codeception\Command\GenerateWPRestApi
    - Codeception\Command\GenerateWPRestController
    - Codeception\Command\GenerateWPRestPostTypeController
    - Codeception\Command\GenerateWPAjax
    - Codeception\Command\GenerateWPCanonical
    - Codeception\Command\GenerateWPXMLRPC
  config:
    tad\WPBrowser\Extension\Copier:
      files:
        .....
params:
  - .env.testing
settings:
  lint: true
  report_useless_tests: true
  colors: true
```

**Suite configuration file**

```yaml
actor: AcceptanceClassicEditorTester
modules:
    enabled:
        - WPDb
        - WPWebDriver
        - \Helper\AcceptanceClassicEditor
        - Asserts
        - WPLoader
        - Sequence
    config:
        WPDb:
            dsn: '%TEST_SITE_DB_DSN%'
            user: '%TEST_SITE_DB_USER%'
            password: '%TEST_SITE_DB_PASSWORD%'
            dump: 'tests/codeception/_data/dump.sql'
            #import the dump before the tests; this means the test site database will be repopulated before the tests.
            populate: true
            # re-import the dump between tests; this means the test site database will be repopulated between the tests.
            cleanup: false
            waitlock: 10
            url: '%TEST_SITE_WP_URL%'
            urlReplacement: true #replace the hardcoded dump URL with the one above
            tablePrefix: '%TEST_SITE_TABLE_PREFIX%'
        WPLoader:
            wpRootFolder: "%WP_ROOT_FOLDER%"
            dbName: "%TEST_DB_NAME%"
            dbHost: "%TEST_DB_HOST%"
            dbUser: "%TEST_DB_USER%"
            dbPassword: "%TEST_DB_PASSWORD%"
            tablePrefix: "%TEST_TABLE_PREFIX%"
            domain: "%TEST_SITE_WP_DOMAIN%"
            adminEmail: "%TEST_SITE_ADMIN_EMAIL%"
            title: "Test"
            plugins: [ ... ]
            activatePlugins: [ ... ]
            isolatedInstall: true
        WPWebDriver:
            url: "%TEST_SITE_WP_URL%"
            adminUsername: "%TEST_SITE_ADMIN_USERNAME%"
            adminPassword: "%TEST_SITE_ADMIN_PASSWORD%"
            adminPath: "%TEST_SITE_WP_ADMIN_PATH%"
            browser: chrome
            restart: true
            window_size: "1280x1200"
            port: 9515
            capabilities:
                browserName: chrome
                chromeOptions:
                    args: [ --disable-gpu", "disable-infobars" ]
                    prefs:
                        download.default_directory: "..."
        Sequence:
            prefix: "uid_"
extensions:
    enabled:
        - Codeception\Extension\RunProcess:
              0: chromedriver --url-base=/wd/hub
              sleep: 3
```

**Describe the bug**

I'm trying to force a DB reload for a specific test, but I can't run the function `importSqlDumpFile` because it returns the error `Call to a member function load() on null`.

```php
$this->importSqlDumpFile(codecept_data_dir('dump.sql'));
```

I'm extending the test actor using a trait, that is why I'm using `$this` instead of calling the actor instance directly.

The function `importSqlDumpFile` on the `WPDb` module is trying to load the driver from the property `$driver`, which is not defined.

**Output**

```
Call to a member function load() on null
 /home/anderson/projects/repos-git/myproject/vendor/lucatume/wp-browser/src/Codeception/Module/WPDb.php:260
 /home/anderson/projects/repos-git/myproject/tests/codeception/_support/_generated/AcceptanceClassicEditorTesterActions.php:37
 /home/anderson/projects/repos-git/myproject/tests/codeception/_support/Steps/Users.php:14
 /tmp/ide-codeception.php:51
```

**To Reproduce**

Try to extend the tester class adding a call to the function `importSqlDumpFile` passing the path of the SQL dump file. You will see the error message described above.

**Suggested fix**

The property `$this->driver` is not defined indeed. But instead of calling it directly, we should call the function `__getDriver()`, which get the correct driver considering the current database being used.

Instead of:

```php
public function importSqlDumpFile($dumpFile = null)
{
    if ($dumpFile !== null) {
        if (!file_exists($dumpFile) || !is_readable($dumpFile)) {
            throw new \InvalidArgumentException("Dump file [{$dumpFile}] does not exist or is not readable.");
        }

        $this->driver->load($dumpFile);

        return;
    }

    if ($this->config['populate']) {
        $this->_cleanup();
        $this->_loadDump();
        $this->populated = true;
    }
}
```

It could be:

```php
public function importSqlDumpFile($dumpFile = null)
{
    if ($dumpFile !== null) {
        if (!file_exists($dumpFile) || !is_readable($dumpFile)) {
            throw new \InvalidArgumentException("Dump file [{$dumpFile}] does not exist or is not readable.");
        }

        $this->_getDriver()->load($dumpFile);

        return;
    }

    if ($this->config['populate']) {
        $this->_cleanup();
        $this->_loadDump();
        $this->populated = true;
    }
}
```
This fixes the issue with the driver not being found (or set). But that displays another error message:

```
Invalid argument supplied for foreach() at vendor/codeception/module-db/src/Codeception/Lib/Driver/Db.php:147
```

The function `load` from the driver expects an array of SQL lines to be executed, not the file path. So, there is an additional fix needed. We could pass the result of `file($dumpFile)` instead of the file path. That will provide the array of SQL that is found in the file.

```php
public function importSqlDumpFile($dumpFile = null)
{
    if ($dumpFile !== null) {
        if (!file_exists($dumpFile) || !is_readable($dumpFile)) {
            throw new \InvalidArgumentException("Dump file [{$dumpFile}] does not exist or is not readable.");
        }

        $this->_getDriver()->load(file($dumpFile));

        return;
    }

    if ($this->config['populate']) {
        $this->_cleanup();
        $this->_loadDump();
        $this->populated = true;
    }
}
```

**Expected behavior**

I would like to see the dump SQL code is loaded from the file into the database after calling that function and passing the path of the dump file.

**Context**

I'm using the `mysql` driver on a MariaDB databse.

**Comment by lucatume** [comment]
@andergmartins thanks for the report and the detailed description.
I'm looking into your PR now and will provide you with feedback shortly.

**Comment by lucatume** [comment]
Fixed by @andergmartins PR and out in [3.0.21](https://github.com/lucatume/wp-browser/releases/tag/3.0.21).
